test/src/test/java/org/corfudb/integration/LogReplicationRoutingQueueIT.java

```diff
@@ -288,6 +308,172 @@ public void testRoutingQueueReplicationWithScopedTxn() throws Exception {
+                            .setReplicationType(Queue.ReplicationType.LOG_ENTRY_SYNC)
+                            .addDestinations(destinationId)
+                            .setOpaquePayload(ByteString.copyFromUtf8("opaquetxn"+i))
+                            .buildPartial();
+                    txn.logUpdateEnqueue(logEntryQ.getStreamUUID(), deltaMessage,
+                            Collections.singletonList(TableRegistry.getStreamIdForStreamTag(CORFU_SYSTEM_NAMESPACE,
+                            LOG_ENTRY_SYNC_QUEUE_TAG_SENDER_PREFIX + destinationId + "_" + DEFAULT_ROUTING_QUEUE_CLIENT)), corfuStore);
+                }
+                txn.commit();
+            }
+        }
+
+        // add end marker to denote end of snapshot sync
+        try(TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+            addMarker(tail, false, DefaultClusterConfig.getSinkClusterIds().get(0), corfuStore, txn);
+            txn.commit();
+        }
+
+
+        // open the recvQ on SINK
+        Table<Queue.CorfuGuidMsg, Queue.RoutingTableEntryMsg, Queue.CorfuQueueMetadataMsg> recvQ = sinkCorfuStores.get(0).openQueue(
+                CORFU_SYSTEM_NAMESPACE, LogReplicationUtils.REPLICATED_RECV_Q_PREFIX + DefaultClusterConfig.getSourceClusterIds().get(0) + "_" + DEFAULT_ROUTING_QUEUE_CLIENT,
+                Queue.RoutingTableEntryMsg.class, TableOptions.builder().schemaOptions(
+                        CorfuOptions.SchemaOptions.newBuilder().addStreamTag(REPLICATED_QUEUE_TAG).build()).build());
+
+        // wait until the recv side count == data written on source
+        while(recvQ.count() < (numLogEntryUpdates + 1)) {
+
+        }
+
+        // the "+1" is for the dummy record that is written at the end of snapshot sync
+        assertThat(recvQ.count()).isEqualTo(numLogEntryUpdates + 1);
+
+
+        AtomicBoolean snapshotSyncData = new AtomicBoolean(true);
+        AtomicInteger countOfSnapshotSyncData = new AtomicInteger(0);
+        AtomicInteger countOfDeltaSyncData = new AtomicInteger(0);
+
+        // validate that the order is maintained.
+        recvQ.entryList().forEach(entry -> {
+            if (snapshotSyncData.get()) {
+                assertEquals(((Queue.RoutingTableEntryMsg) entry.getEntry()).getReplicationType(), Queue.ReplicationType.SNAPSHOT_SYNC);
+                countOfSnapshotSyncData.incrementAndGet();
+            } else {
+                assertEquals(((Queue.RoutingTableEntryMsg) entry.getEntry()).getReplicationType(), Queue.ReplicationType.LOG_ENTRY_SYNC);
+                countOfDeltaSyncData.incrementAndGet();
+            }
+            // flip the flag after all the snapshot sync data is received. From now, there should only be delta sync data
+            if(countOfSnapshotSyncData.get() == numLogEntryUpdates/2 && countOfDeltaSyncData.get() == 0) {
+                snapshotSyncData.set(false);
+            }
+
+        });
+    }
+
+    class SnapshotRequestListener implements StreamListener {
```

Comment:
can the existing SnapSyncRequestor be re-used here?

Reply:
not really, because we want to control the writes on the source.. If we want to reuse the snapSyncRequestor, we'll need to have a no-op snapSyncProvider.